slides/11_State_Management.html

```diff
@@ -78,30 +79,33 @@ <h3>Step by Step</h3>
           </section>
 
           <section>
-            <pre>
-                    <code class="hljs tsx" data-trim>
-                        <script type="text/template">
+            <pre><code class="hljs tsx" data-trim><script type="text/template">
             import { createContext, useContext } from "react";
 
-            type YourContextValue = {
-              user: string;
+            type ContextType = {
+              name: string;
+              setName: (name: string) => void;            
             };
             
-            const UserContext = createContext<YourContextValue | undefined>(undefined);
+            const UserContext = createContext<ContextType | undefined>(undefined);
             
             function Parent() {
-              return (<UserContext.Provider value={{ user: "joe" }}>
-                <Child />
-              </UserContext.Provider>)
+              const [name, setName] = useState("Joe");
+              return (
+                <UserContext.Provider value={{ name, setName }}>
+                  <Child />
+                </UserContext.Provider>)
             }
 
             function Child() {
               const context = useContext(UserContext);
-              return <p>{context?.user}</p>; // joe
+              const updateName = () => context?.setName("Jane")
```

Comment:
I would recommend to the students to use the fail fast principle here and do a check if context is defined. If it is not defined, an error should be thrown. This way the error is apparent if Child is used outside of the Provider and the conditional access operator in this line and below is not needed.

Reply:
I didn't put an error check here as we show the clean solution of wrapping `useContext` into your own custom hook including error check/handling on one of the following solution slides. Shall we still do a simple check here?

Reply:
Sorry for the long delay, just see this comment now. Yeah, it's fine then like this :-)